tests/backend/test_bosonic_backend.py

```diff
@@ -0,0 +1,566 @@
+        l1 = [1, 2]
+        l2 = [3, 4, 5]
+        list_compare = [3, 4, 5, 6, 8, 10]
+        assert np.allclose(list_compare, bosonic.kron_list([l1, l2]))
+
+
+class TestBosonicCatStates:
+    r"""Tests cat state method of the BosonicBackend class."""
+
+    @pytest.mark.parametrize("alpha", ALPHA_VALS)
+    @pytest.mark.parametrize("phi", PHI_VALS)
+    def test_cat_complex(self, alpha, phi):
+        r"""Checks the complex cat state representation."""
+        prog = sf.Program(1)
+        with prog.context as q:
+            sf.ops.Catstate(alpha, phi) | q[0]
+
+        backend = bosonic.BosonicBackend()
+        backend.run_prog(prog)
+        state = backend.state()
+
+        if alpha != 0:
+            # Check shapes
+            assert state.num_weights == 4
+            assert state.weights().shape == (4,)
+            assert np.allclose(sum(state.weights()), 1)
+            assert state.means().shape == (4, 2)
+            assert state.covs().shape == (4, 2, 2)
+            covs_compare = np.tile(np.eye(2) * sf.hbar / 2, (4, 1, 1))
+            assert np.allclose(state.covs(), covs_compare)
+
+            # Weights not real if phi != 0 or 1
+            if phi % 1 == 0:
+                assert np.allclose(state.weights().real, state.weights())
+            else:
+                assert not np.allclose(state.weights().real, state.weights())
+            # Covs should be real, means complex
+            assert not np.allclose(state.means().real, state.means())
+            assert np.allclose(state.covs().real, state.covs())
+        else:
+            assert state.num_weights == 1
+            assert state.weights().shape == (1,)
+            assert np.allclose(sum(state.weights()), 1)
+            assert state.means().shape == (1, 2)
+            assert state.covs().shape == (1, 2, 2)
+            covs_compare = np.tile(np.eye(2) * sf.hbar / 2, (1, 1, 1))
+            assert np.allclose(state.covs(), covs_compare)
+
+    @pytest.mark.parametrize("alpha", ALPHA_VALS)
+    @pytest.mark.parametrize("phi", PHI_VALS)
+    def test_cat_real(self, alpha, phi):
+        r"""Checks the real cat state representation."""
+        # Check that low cutoff and low D produce fewer weights when alpha !=0
+        prog = sf.Program(1)
+        with prog.context as q:
+            sf.ops.Catstate(alpha, phi, cutoff=1e-6, desc="real", D=1) | q[0]
+
+        backend = bosonic.BosonicBackend()
+        backend.run_prog(prog)
+        state = backend.state()
+        num_weights_low = state.num_weights
+
+        assert np.allclose(sum(state.weights()), 1)
+        assert state.means().shape == (num_weights_low, 2)
+        assert state.covs().shape == (num_weights_low, 2, 2)
+
+        # Weights, means and covs should be real
+        assert np.allclose(state.weights().real, state.weights())
+        assert np.allclose(state.means().real, state.means())
+        assert np.allclose(state.covs().real, state.covs())
+
+        prog = sf.Program(1)
```

Comment:
This seems to be a separate test case, worth creating a separate test method.

Reply:
Done.